core/src/avm2/globals/flash/geom/point.rs

```diff
@@ -0,0 +1,416 @@
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        let (our_x, our_y) = coords(&mut this, activation)?;
+
+        return create_point(activation, (our_x, our_y));
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `copyFrom`
+pub fn copy_from<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        if let Some(other) = args.get(0) {
+            let mut other_obj = other.coerce_to_object(activation)?;
+            let (their_x, their_y) = coords(&mut other_obj, activation)?;
+
+            set_coords(&mut this, activation, (their_x, their_y))?;
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `distance`
+pub fn distance<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(first) = args.get(0) {
+        let mut first_object = first.coerce_to_object(activation)?;
+        if let Some(second) = args.get(1) {
+            let mut second_obj = second.coerce_to_object(activation)?;
+            let (our_x, our_y) = coords(&mut first_object, activation)?;
+            let (their_x, their_y) = coords(&mut second_obj, activation)?;
+
+            return Ok(((our_x - their_x).powf(2.0) + (our_y - their_y).powf(2.0))
+                .sqrt()
+                .into());
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `equals`
+pub fn equals<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(mut this) = this {
+        if let Some(other) = args.get(0) {
+            let mut other_obj = other.coerce_to_object(activation)?;
+
+            let our_x =
+                this.get_property(this, &QName::new(Namespace::public(), "x"), activation)?;
+            let our_y =
+                this.get_property(this, &QName::new(Namespace::public(), "y"), activation)?;
+
+            let their_x = other_obj.get_property(
+                other_obj,
+                &QName::new(Namespace::public(), "x"),
+                activation,
+            )?;
+            let their_y = other_obj.get_property(
+                other_obj,
+                &QName::new(Namespace::public(), "y"),
+                activation,
+            )?;
+
+            return Ok((our_x == their_x && our_y == their_y).into());
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `interpolate`
+pub fn interpolate<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    _this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if args.len() < 3 {
+        return create_point(activation, (f64::NAN, f64::NAN));
+    }
```

Comment:
TODO: argument counts need to be checked by the VM itself, which will throw an error if a function is called with the incorrect arguments (we don't do this yet).